spec/services/doc_auth/lexis_nexis/requests/true_id_request_spec.rb

```diff
@@ -24,34 +28,74 @@
       image_source: image_source,
       user_uuid: applicant[:uuid],
       uuid_prefix: applicant[:uuid_prefix],
+      selfie_image: selfie_image,
+      liveness_checking_enabled: liveness_checking_enabled,
     )
   end
 
   shared_examples 'a successful request' do
     it 'uploads the image and returns a successful result' do
-      request_stub = stub_request(:post, full_url).to_return(body: response_body, status: 201)
+      include_liveness = liveness_checking_enabled && !selfie_image.nil?
+      request_stub_liveness = stub_request(:post, full_url).with do |request|
+        JSON.parse(request.body, symbolize_names: true)[:Document][:Selfie].present?
+      end.to_return(body: response_body(include_liveness), status: 201)
+      request_stub = stub_request(:post, full_url).with do |request|
+        !JSON.parse(request.body, symbolize_names: true)[:Document][:Selfie].present?
+      end.to_return(body: response_body(include_liveness), status: 201)
 
       response = subject.fetch
 
       expect(response.success?).to eq(true)
       expect(response.errors).to eq({})
       expect(response.exception).to be_nil
-      expect(request_stub).to have_been_requested
+      if include_liveness
+        expect(request_stub_liveness).to have_been_requested
+      else
+        expect(request_stub).to have_been_requested
+      end
     end
   end
 
-  context 'with acuant image source' do
-    let(:workflow) { 'test_workflow' }
-    let(:image_source) { DocAuth::ImageSources::ACUANT_SDK }
+  context 'with liveness_checking_enabled as false' do
+    let(:liveness_checking_enabled) { false }
+    context 'with acuant image source' do
+      let(:workflow) { 'test_workflow' }
+      let(:image_source) { DocAuth::ImageSources::ACUANT_SDK }
+      it_behaves_like 'a successful request'
 
-    it_behaves_like 'a successful request'
+      context 'with no selfie image' do
+        let(:selfie_image) { nil }
+        it_behaves_like 'a successful request'
+      end
+    end
+    context 'with unknown image source' do
+      let(:workflow) { 'test_workflow_cropping' }
+      let(:image_source) { DocAuth::ImageSources::UNKNOWN }
+
+      it_behaves_like 'a successful request'
+    end
   end
 
-  context 'with unknown image source' do
-    let(:workflow) { 'test_workflow_cropping' }
-    let(:image_source) { DocAuth::ImageSources::UNKNOWN }
+  context 'with liveness_checking_enabled as true' do
+    let(:liveness_checking_enabled) { true }
+    context 'with acuant image source' do
+      let(:workflow) { 'test_workflow_liveness' }
+      let(:image_source) { DocAuth::ImageSources::ACUANT_SDK }
 
-    it_behaves_like 'a successful request'
+      it_behaves_like 'a successful request'
+
+      context 'with no selfie image' do
+        let(:selfie_image) { nil }
+        let(:workflow) { 'test_workflow' }
+        it_behaves_like 'a successful request'
+      end
```

Comment:
Checking my own understanding: Does this mean for the time being, even if liveness is enabled, but there is no image, flow can continue on?

Reply:
Hmm [this line](https://github.com/18F/identity-idp/pull/9561/files#diff-a3eac8c46b97505dc1b71fe1a57c05f0027658b4b531a66678d1d0b1312ac055R8) seems to contradict my comment here. I must be misunderstanding what this test does. Do you mind clarifying?

Reply:
@night-jellyfish yes, it allow the external request keep going, even if liveness is enabled but no selfie is available, it will fall back to non-liveness workflow.

The form validation check is there since we will enable liveness check in dev for all cases. At lowest level, this gives us flexibility to deal with possible testing bucket split, SP specific requirements about liveness check etc, also avoid changing existing tests(especially those feature tests) while things are in fluid state.

Reply:
Got it, thank you for the explanation!